solr/core/src/test/org/apache/solr/cloud/TestPullReplica.java

```diff
@@ -528,17 +529,21 @@ private void waitForNumDocsInAllActiveReplicas(int numDocs) throws IOException,
   }
 
   private void waitForNumDocsInAllReplicas(int numDocs, Collection<Replica> replicas) throws IOException, SolrServerException, InterruptedException {
-    waitForNumDocsInAllReplicas(numDocs, replicas, "*:*");
+    waitForNumDocsInAllReplicas(numDocs, replicas, "*:*", null, null);
   }
 
-  private void waitForNumDocsInAllReplicas(int numDocs, Collection<Replica> replicas, String query) throws IOException, SolrServerException, InterruptedException {
+  static void waitForNumDocsInAllReplicas(int numDocs, Collection<Replica> replicas, String query, String user, String pass) throws IOException, SolrServerException, InterruptedException {
     TimeOut t = new TimeOut(REPLICATION_TIMEOUT_SECS, TimeUnit.SECONDS, TimeSource.NANO_TIME);
     for (Replica r:replicas) {
       try (HttpSolrClient replicaClient = getHttpSolrClient(r.getCoreUrl())) {
         while (true) {
           try {
+            QueryRequest req = new QueryRequest(new SolrQuery(query));
```

Comment:
nit: should be able to create the object once outside of the loop?

Reply:
I just adapted the code that was already there to be able to pass credentials and left it largely as-is to minimize changes for this particular issue. I'm not taking on a full refactor of a disabled test in this PR.